main.go

```diff
@@ -53,21 +50,32 @@ func main() {
 		}
 	}
 
-	var a *authenticator
+	var src credentialSource
 	if *domain != "" {
-		fmt.Printf("Password (for %s\\%s): ", *domain, *username)
-		buf, err := term.ReadPassword(int(os.Stdin.Fd()))
-		fmt.Println()
-		if err != nil {
-			log.Fatalf("Error reading password from stdin: %v", err)
-		}
-		a = &authenticator{domain: *domain, username: *username, hash: getNtlmHash(buf)}
-	} else if getCredentialsFromKeyring != nil {
+		src = fromTerminal().forUser(*domain, *username)
+	} else if value, ok := os.LookupEnv("NTLM_CREDENTIALS"); ok {
```

Comment:
I would consider making this:
```
} else if value := os.Getenv("NTLM_CREDENTIALS"); value != "" {
```

That way, you can easily launch alpaca to make it not use the envvar without actually unsetting it in your shell by running:
```
$ NTLM_CREDENTIALS= ./alpaca
```

An empty envvar is as good as an unset envar, so no point even trying to use it I think.